Assets/Plugins/CountlySDK/Countly.cs

```diff
@@ -216,18 +218,35 @@ public async void Init(CountlyConfiguration configuration)
             Session = new SessionCountlyService(Configuration, Events, _push, requests, Location, Consents, eventNumberInSameSessionHelper);
 
             CrashReports = new CrashReportsCountlyService(Configuration, requests);
-
-            Device = new DeviceIdCountlyService(Session, requests, Events, countlyUtils);
             Initialization = new InitializationCountlyService(Configuration, Location, Consents, Session);
-
             RemoteConfigs = new RemoteConfigCountlyService(Configuration, requests, countlyUtils, configDao);
 
             StarRating = new StarRatingCountlyService(Events);
             UserDetails = new UserDetailsCountlyService(requests, countlyUtils);
             Views = new ViewCountlyService(Configuration, Events);
+            Device = new DeviceIdCountlyService(Configuration, Session, requests, Events, countlyUtils);
             _inputObserver = InputObserverResolver.Resolve();
+
+            RegisterServicesToDeviceService();
         }
 
+        private void RegisterServicesToDeviceService() {
+            _listeners.Clear();
+
+            _listeners.Add(Consents);

```

Comment:
All of these listeners should be added to the array independently of the device ID service.
Either rename this function to something more generic or move all of the "_listeners.Add(...)" outside of this function.

Reply:
"All of these listeners should be added to the array independently of the device ID service."

We have to define the size of the array at the start,
independently of the device ID service --> need more words to understand.
Do you mean DeviceId Service tell countly.cs class on device id change and countly.cs will tell all other listeners(services)?